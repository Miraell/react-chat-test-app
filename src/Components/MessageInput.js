import React, { useState } from 'react';

function MessageInput({ setMessage, isMessageSending }) {

    const [countSymbolsLeft, setCountSymbolsLeft] = useState(250);

    const handleChange = (event) => {
        setMessage(event.target.value);
        setCountSymbolsLeft(250 - event.target.value.length);
    }

    if (isMessageSending === true) {
        return (
            <div>Sending...</div>
        );
    }
    return (
        <div>
        You have { countSymbolsLeft } symbols left.
        <input
            onChange={handleChange}
            type="text"
        />
        </div>
    );
}

export default MessageInput;
