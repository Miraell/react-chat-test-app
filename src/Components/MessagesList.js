import React from 'react';

function messagesList(props) {
    const messages = props.messages ? props.messages.map((message, index) =>
        <li key={index}>
            <b>{message.user}</b> says: <span className="bubble">{message.text}</span>
        </li>
    ) : 'No messages';
    return (
        <div className="messages-wrapper">
            {messages}
        </div>
    );
}

export default messagesList;
