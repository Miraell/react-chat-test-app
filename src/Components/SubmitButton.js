import React from 'react';

export default function SubmitButton({isMessageSending, onClick}) {

    if (isMessageSending === true) {
        return null;
    }
    return <button onClick={onClick}>Send</button>
}
