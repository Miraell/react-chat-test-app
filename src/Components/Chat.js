import React from 'react';
import MessagesList from './MessagesList';
import MessageInput from './MessageInput';
import SubmitButton from './SubmitButton';
import { connect } from 'react-redux';
import actions from '../redux/actions';

const DEFAULT_USER_NAME = 'Dmitry';

export class Chat extends React.Component {

    setMessage(inputMessage) {
        console.log('test');
        this.setState({ inputMessage });
    }

    clickButton() {
        this.props.sendMessage({
            user: DEFAULT_USER_NAME,
            text: this.state.inputMessage,
        });
    }

    render() {
        return (
            <div>
                <MessagesList messages={this.props.messages} /><br />
                <form>
                    <MessageInput
                        setMessage={this.setMessage.bind(this)}
                        isMessageSending={this.props.isMessageSending}
                    />
                    <SubmitButton
                        onClick={this.clickButton.bind(this)}
                        isMessageSending={this.props.isMessageSending}
                    />
                </form>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    messages: state.messages,
    isMessageSending: state.isMessageSending,
});

const mapDispatchToProps = (dispatch) => {
    return {
        sendMessage: (message) => {
            dispatch(actions.sendMessage(message));
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(Chat);
