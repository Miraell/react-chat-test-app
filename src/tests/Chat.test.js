import React from 'react';
import { Chat } from '../Components/Chat';
import renderer from 'react-test-renderer';

test('Chat snapshot test', () => {
    const component = renderer.create(
        <Chat />
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
