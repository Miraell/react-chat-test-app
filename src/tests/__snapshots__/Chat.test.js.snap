// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat snapshot test 1`] = `
<div>
  <div
    className="messages-wrapper"
  >
    No messages
  </div>
  <br />
  <form>
    <div>
      You have 
      250
       symbols left.
      <input
        onChange={[Function]}
        type="text"
      />
    </div>
    <button
      onClick={[Function]}
    >
      Send
    </button>
  </form>
</div>
`;
